include/pybind11/pybind11.h

```diff
@@ -203,6 +199,8 @@ class cpp_function : public function {
                 a.descr = strdup(a.value.attr("__repr__")().cast<std::string>().c_str());
         }
 
+        rec->is_constructor = !strcmp(rec->name, "__init__") || !strcmp(rec->name, "__setstate__");
```

Comment:
It was part of an earlier commit that was since dropped to fix the constructor signature when the first argument was a `handle`, but that got replaced with `v_h_proxy<Cpp<Class>>`.  I think you want me to go back to that, though.